### PR TITLE
Flipkart clone added

### PR DESCRIPTION
**Pull request by harshilsharmaa**
Flipkart clone project with all steps and challenges added.

### CODE REVIEW

**Review by Pranavtechie — CHANGES_REQUESTED** [review]
Your Boilerplate repo does not contain `.cdmrc` configuration file. 

Please refer to [https://teach.codedamn.com/docs/concepts/cdmrc](https://teach.codedamn.com/docs/concepts/cdmrc) add the file in your starter files repo

**Comment by harshilsharmaa** [comment]
Updated and .cdmrc configuration file added.

File: flipkart-clone/step3/challenges.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: 'Add Pagination functionality and API testing'
+stepId: step3
+stepBreakDown:
+    - text: 'Make logic for pagination in product.js file in controllers folder'
+    - text: 'Pass page and limit as query parameters in products route'
+    - text: 'Tesr your APIs using postman or any other API testing tool'
```
Comment (Pranavtechie) [inline flipkart-clone/step3/challenges.yml:6]: Typo at `Tesr`

**Review by Pranavtechie — CHANGES_REQUESTED** [review]
Hey @harshilsharmaa, you've done a great job in breaking down the project into simple steps. 

You can work on making the Instructions more specific and elaborate such that they help the user in completing the challenges.

File: flipkart-clone/step3/challenges.yml (hunk 1 above)
Comment (harshilsharmaa) [reply flipkart-clone/step3/challenges.yml]: Typo Corrected.

**Comment by harshilsharmaa** [comment]
Updated Instructions to get better understanding to the users.

**Comment by harshilsharmaa** [comment]
Please review this.

File: flipkart-clone/step1/Instructions.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - 200 for success
+    - 400 for bad request
+    - 404 for not found
+    - 500 for internal server error
+    - etc. (You can take reference from [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status))
+
+#### Make some basic APIs:
+- Get all products
+- Get Products by category
+- Add Products
```
Comment (Pranavtechie) [inline flipkart-clone/step1/Instructions.md:20]: I couldn't find the details of the product, where they being fetched from

File: flipkart-clone/step10/Instructions.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+## Show dynamic content
+
+1. use hooks like useState and useEffect etc.
```
Comment (Pranavtechie) [inline flipkart-clone/step10/Instructions.md:3]: Please provide the context to the user in the Instructions file as well, regarding where to do.

File: flipkart-clone/step10/challenges.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+name: 'Show the data on components'
```
Comment (Pranavtechie) [inline flipkart-clone/step10/challenges.yml:1]: Rename step to "Displaying Data"

**Review by Pranavtechie — CHANGES_REQUESTED** [review]

**Comment by Pranavtechie** [comment]
@harshilsharmaa are you working on the requested changes?

**Comment by harshilsharmaa** [comment]
> @harshilsharmaa are you working on the requested changes?

Yes, Iam working on it.

**Pull request closed without merging**